### PR TITLE
Log testhost standard error to diag log file

### PR DESCRIPTION
**Pull request by smadala**

### CODE REVIEW

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs [hunk 1]
```diff
@@ -101,19 +101,29 @@ internal DefaultTestHostManager(Architecture architecture, Framework framework,
...
 
         /// <summary>
         /// Gets callback to read from process error stream
         /// </summary>
         private Action<object, string> ErrorReceivedCallback => (process, data) =>
         {
-            var exitCode = 0;
             if (!string.IsNullOrEmpty(data))
             {
+                EqtTrace.Warning("Test host standard error line: {0}", data);
```
Comment (codito) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: Please add a comment why this is required?

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs [hunk 2]
```diff
@@ -101,19 +101,29 @@ internal DefaultTestHostManager(Architecture architecture, Framework framework,
         private Action<object> ExitCallBack => (process) =>
         {
             var exitCode = 0;
+            var testHostProcessStdErrorStr = this.testHostProcessStdError.ToString();
```
Comment (codito) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: This logic is duplicated across Default and DotnetTestHostManagers. Should extract?

File: test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs [hunk 3]
```diff
@@ -94,16 +96,58 @@ public void WorkingDirectoryIsSourceDirectory(string runnerFramework, string tar
...
                 this.BuildMultipleAssemblyPath("SimpleTestProject3.dll").Trim('\"');
             var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), string.Empty, this.FrameworkArgValue);
-            arguments = string.Concat(arguments, " /tests:ExitWithStdErrorMessageTest");
+            arguments = string.Concat(arguments, " /tests:ExitwithUnhandleException");
+            arguments = string.Concat(arguments, $" /diag:{diagLogFilePath}");
+
+            this.InvokeVsTest(arguments);
+            var errorMessage = string.Empty;
+
+            if (targetFramework.StartsWith("netcoreapp2."))
```
Comment (codito) [inline test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs]: :)

File: test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs [hunk 4]
```diff
@@ -94,16 +96,58 @@ public void WorkingDirectoryIsSourceDirectory(string runnerFramework, string tar
...
 
+            var diagLogFilePath = Path.Combine(Path.GetTempPath(), $"std_error_log_{Guid.NewGuid()}.txt");
+            File.Delete(diagLogFilePath);
+
             var assemblyPaths =
                 this.BuildMultipleAssemblyPath("SimpleTestProject3.dll").Trim('\"');
             var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), string.Empty, this.FrameworkArgValue);
-            arguments = string.Concat(arguments, " /tests:ExitWithStdErrorMessageTest");
+            arguments = string.Concat(arguments, " /tests:ExitwithUnhandleException");
+            arguments = string.Concat(arguments, $" /diag:{diagLogFilePath}");
```
Comment (codito) [inline test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs]: We shouldn't enable `diag` for this scenario. This should ensure everything shows up in console and VS without any special setting.

File: test/Microsoft.TestPlatform.TestUtilities/FileAssert.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+namespace TestPlatform.TestUtilities
+{
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System.IO;
+
+    public static class FileAssert
+    {
+        private const string StringHighlighter = "\"\"\"";
+
+        public static void Contains(string filePath, params string[] substrs)
```
Comment (codito) [inline test/Microsoft.TestPlatform.TestUtilities/FileAssert.cs]: We should use the new `Assert.That` extensibility :) cc: @AbhitejJohn

File: test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Microsoft.TestPlatform.TestHostProvider.UnitTests.csproj [hunk 6]
```diff
@@ -1,38 +1,38 @@
...
-    <Reference Include="System.Runtime" />
-    <Reference Include="System" />
-    <Reference Include="Microsoft.CSharp" />
-    <Reference Include="System.Xml" />
-  </ItemGroup>
-  <ItemGroup>
-    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
-  </ItemGroup>
-  <Import Project="$(TestPlatformRoot)scripts\build\TestPlatform.targets" />
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (codito) [inline test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Microsoft.TestPlatform.TestHostProvider.UnitTests.csproj]: Not sure why this file shows up.

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs [hunk 7]
```diff
@@ -101,19 +101,29 @@ internal DefaultTestHostManager(Architecture architecture, Framework framework,
...
+
             this.processHelper.TryGetExitCode(process, out exitCode);
 
-            this.OnHostExited(new HostProviderEventArgs(this.testHostProcessStdError.ToString(), exitCode, (process as Process).Id));
+            if (exitCode != 0)
+            {
+                EqtTrace.Error("Test host exited with error: {0}", this.testHostProcessStdError);
+                this.messageLogger.SendMessage(TestMessageLevel.Error, testHostProcessStdErrorStr);
```
Comment (codito) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: Please validate that error shows up in both VS and CLI.

**Comment by codito** [comment]
Can this be inserted to dotnet-cli soon?

**Review by codito — APPROVED** [review]

**Comment by smadala** [comment]
@dotnet-bot please test Windows_NT / Release Build

**Review by mayankbansal018 — APPROVED** [review]

**Pull request merged**